### DISCUSSION TITLE
[Bug] Program In Administrator Path Override Program Installed By Scoop

### DISCUSSION THREAD

**Discussion by ghost** [ScoopInstaller/Scoop#5023] [Help and Questions] [answered]
#### Current Behavior
some programs are already in \Windows\System32 folder like curl and some others are internals to cmd.exe like mkdir, these override scoop installed programs how can I fix this, unfortunately, I wasn't available to find a guide on this topic, sorry if it's kinda irrelevant

```
> where curl
C:\Windows\System32\curl.exe
C:\Users\Sina\.scoop\shims\curl.exe

> which curl
C:\WINDOWS\SYSTEM32\curl.EXE
```
or worst case
```
> where mkdir
C:\Users\Sina\.scoop\shims\mkdir.exe

> mkdir /?
[IT'S HELP MESSAGE]

> C:\Users\Sina\.scoop\shims\mkdir.exe /?
"C:/Users/Sina/.scoop/apps/coreutils/current/bin/mkdir.exe": cannot create directory `/?': No such file or directory
```

### System details

**Windows version:** 11

**OS architecture:** 64bit

**PowerShell version:** 5.1.22000.653

**Comment by rashil2000** [comment, marked as answer]
System's PATH will always have priority over User's PATH. For curl, have you tried installing it globally?

For mkdir, nothing can be done as mkdir is an CMD built-in and it will always prioritize builtins. Alternative is to use full path, or create an alias to mkdir with `scoop shim add mkdir-s C:\Users\Sina\.scoop\shims\mkdir.exe`

**Comment by niheaven** [comment]
The system curl always has highest priority and this could not be changed (I've tried all my ways) 😢

**Reply by hrxn** [reply]
Not sure what the default is with CMD, but if you use PowerShell this can be changed easily.

Because aliases have a higher command precedence, you can simply set an alias in your PS profile that points to a specific binary of curl, like this:
`New-Alias -Name curl -Value 'C:\Program Files\Whatever\Whocares\curl.exe'`
